sormas-backend/src/test/java/de/symeda/sormas/backend/immunization/ImmunizationFacadeEjbPseudonymizationTest.java

```diff
@@ -0,0 +1,217 @@
+		loginWith(nationalAdmin);
+
+		// immunization within limited user's jurisdiction
+		PersonDto person1 = creator.createPerson("John", "Doe");
+		final ImmunizationDto immunization1 = creator.createImmunization(
+			Disease.CORONAVIRUS,
+			person1.toReference(),
+			nationalUser.toReference(),
+			ImmunizationStatus.ACQUIRED,
+			MeansOfImmunization.VACCINATION,
+			ImmunizationManagementStatus.COMPLETED,
+			rdcf1);
+		immunization1.setAdditionalDetails("confidential details");
+		getImmunizationFacade().save(immunization1);
+
+		// immunization outside limited user's jurisdiction
+		PersonDto person2 = creator.createPerson("Max", "MUstermann");
+		final ImmunizationDto immunization2 = creator.createImmunization(
+			Disease.CORONAVIRUS,
+			person2.toReference(),
+			nationalUser.toReference(),
+			ImmunizationStatus.ACQUIRED,
+			MeansOfImmunization.VACCINATION,
+			ImmunizationManagementStatus.COMPLETED,
+			rdcf2);
+		immunization2.setAdditionalDetails("confidential details second");
+		getImmunizationFacade().save(immunization2);
+
+		loginWith(districtUser1);
+		ImmunizationDto testImmunization3 = getImmunizationFacade().getByUuid(immunization1.getUuid());
+		assertThat(testImmunization3.isPseudonymized(), is(false));
+		assertThat(testImmunization3.getAdditionalDetails(), is("confidential details"));
+		ImmunizationDto testImmunization3Second = getImmunizationFacade().getByUuid(immunization2.getUuid());
+		assertThat(testImmunization3Second.isPseudonymized(), is(true));
+		assertThat(testImmunization3Second.getAdditionalDetails(), is(emptyString()));
+	}
+
+	@Test
+	public void testPseudonymizedGetByUuidWithLimitedUser() {
+
+		// deactivate AUTOMATIC_RESPONSIBILITY_ASSIGNMENT in order to assign the limited user to a case from outside jurisdiction
+		FeatureConfigurationIndexDto featureConfiguration =
+			new FeatureConfigurationIndexDto(DataHelper.createUuid(), null, null, null, null, null, true, null);
+		getFeatureConfigurationFacade().saveFeatureConfiguration(featureConfiguration, FeatureType.CASE_SURVEILANCE);
+
+		executeInTransaction(em -> {
+			Query query = em.createQuery("select f from featureconfiguration f");
+			FeatureConfiguration singleResult = (FeatureConfiguration) query.getSingleResult();
+			HashMap<FeatureTypeProperty, Object> properties = new HashMap<>();
+			properties.put(FeatureTypeProperty.AUTOMATIC_RESPONSIBILITY_ASSIGNMENT, false);
+			singleResult.setProperties(properties);
+			em.persist(singleResult);
+		});
+
+		loginWith(nationalAdmin);
+
+		// immunization within limited user's jurisdiction
+		PersonDto person1 = creator.createPerson("John", "Doe");
+		final ImmunizationDto immunization1 = creator.createImmunization(
+			Disease.CORONAVIRUS,
+			person1.toReference(),
+			nationalUser.toReference(),
+			ImmunizationStatus.ACQUIRED,
+			MeansOfImmunization.VACCINATION,
+			ImmunizationManagementStatus.COMPLETED,
+			rdcf1);
+		immunization1.setAdditionalDetails("confidential details");
```

Comment:
please use the creator method with a consumer so you don't need to update the object right after creation

Reply:
changed the creator method to the one that has a consumer